### PR TITLE
Add drilled tile

### PR DESCRIPTION
**Pull request by mkofler96**
# Overview
added new microtile

## Addressed issues
.

## Showcase
A "short" example to highlight the (new) feature
```
# %%
import splinepy as sp
import gustaf as gus
import numpy as np
import vedo

# comment this line to visualize within the notebook (limited functionalities)
vedo.settings.default_backend = "vtk"

# %%
microstructure = sp.microstructure.Microstructure()
microstructure.deformation_function = sp.Bezier(
    degrees=[1, 1],
    control_points=[[0, 0], [0.5,0], [0, 1], [0.5, 1]],
)

angles = np.linspace(-np.pi/2+0.01,np.pi/2-0.01, 10)

tiles = []
for a in angles:
    parameters = np.ones((1,1))*a
    my_microtile = sp.microstructure.tiles.DrilledCrossTile1D()
    showable_tile = my_microtile.create_tile(parameters=parameters)
    tiles.append(showable_tile[0])
gus.show(*tiles, control_points=False)

# %%
# Parameter spline
para_s = sp.BSpline(
    degrees=[1, 1],
    knot_vectors=[[0, 0, 1, 1], [0, 0,0.5, 1, 1]],
    control_points=[[1], [0], [0], [0] , [1], [0]],
)


def pf_curved_tile(x):
    """
    Parametrization Function (determines thickness)
    """
    
    return np.pi/2*para_s.evaluate(x)

# %%
ms = sp.microstructure.Microstructure()
ms.deformation_function = sp.Bezier(
    degrees=[1, 1],
    control_points=[[0, 0], [0.5,0], [0, 1], [0.5, 1]],
)

my_microtile = sp.microstructure.tiles.DrilledCrossTile1D()
showable_tile = my_microtile.create_tile()

#ms.microtile = showable_tile[0]
ms.microtile = my_microtile
ms.tiling = [8, 16]
ms.parametrization_function = pf_curved_tile
ms.show(control_points=False, title="2D Lattice Microstructure", lighting="off")



```

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

**Comment by jzwar** [comment]
Please rebase onto branch `cu-tile-names`, or wait for it to merge.

The name of the file should be in `snake_case`, please. Also, consider naming the tile
`DrilledCrossBeams` or `TwistedCrossBeams` to avoid confusion...

File: splinepy/microstructure/tiles/drilled_cross_beams.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+        # set to default if nothing is given
+        if parameters is None:
+            self._logd(
+                "Tile request is not parametrized, setting default Pi/8"
+            )
+            parameters = np.array([[np.pi / 8]])
+        else:
+            if not (
+                np.all(parameters >= -np.pi) and np.all(parameters < np.pi)
```
Comment (jzwar) [inline splinepy/microstructure/tiles/drilled_cross_beams.py:44]: These bounds are too wide. 180 degrees would just result in a tangled line.

File: splinepy/microstructure/tiles/drilled_cross_beams.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            derivatives = []
+        else:
+            n_derivatives = 0
+            derivatives = None
+
+        splines = []
+        for i_derivative in range(n_derivatives + 1):
+            # Constant auxiliary values
+            if i_derivative == 0:
+                alpha = parameters[0, 0] + np.pi / 4  # parameters.shape == [1]
```
Comment (jzwar) [inline splinepy/microstructure/tiles/drilled_cross_beams.py:62]: Why the offset of 45 degs? also the commentary is wrong. the shape is two-dimensional

File: splinepy/microstructure/tiles/drilled_cross_beams.py
Comment (jzwar) [inline splinepy/microstructure/tiles/drilled_cross_beams.py:1]: This entire file looks like a duplicate! Please check this and re request review once this is properly rebased!

**Review by jzwar — CHANGES_REQUESTED** [review]
Nice addition, but please check if you created a duplicate file here. It looks unintentional. 

Also rebase required after tile renaming and include statements with underscore.

**Comment by mkofler96** [comment]
moved to #316

**Pull request closed without merging**